Detta är ett väldigt rudimentärt analytiskt program som arbetat med statestik och linjär regrition.
Det som är viktigt är inte vad som syns här utan det som är under huven. Men det vi kan se hör är att den inte bara läser in den tilldelade data.

In [1]:
from Labb_statestik import LinearRegression
import numpy as np

data = np.genfromtxt ("C:\\Users\\matti\\Downloads\\housing.csv", delimiter=',',skip_header=1, dtype=str)

X_raw = data [:, [2,3,4]]
y_raw = data[:,8]



In [3]:
mask = ((X_raw !='').all(axis=1) & (y_raw !=''))
X = X_raw[mask].astype(float)
y = y_raw[mask].astype(float)


In [4]:
model = LinearRegression()
model.fit(X,y)

In [5]:
model.summary()

{'variables': ['Intercept', 'x0', 'x1', 'x2'],
 'coefficients': array([ 1.39853372e+05,  1.71931313e+03,  3.88285878e+01, -1.57268229e+02]),
 'std_error': array([2.55339652e+03, 6.56847704e+01, 9.77978459e-01, 4.99378787e+00]),
 't': array([ 54.77150577,  26.17521712,  39.70290705, -31.49277317]),
 'p': array([0.00000000e+000, 1.42654168e-148, 0.00000000e+000, 1.29609567e-212]),
 'CI_low': array([ 1.34848511e+05,  1.59056571e+03,  3.69116717e+01, -1.67056453e+02]),
 'CI_high': array([ 1.44858234e+05,  1.84806054e+03,  4.07455040e+01, -1.47480004e+02]),
 'R2': 0.08941412286766037,
 'Adj_R2': 0.08928040327142961,
 'F': 668.6688068770183,
 'F_p': np.float64(0.0),
 'RMSE': 110162.13848873758}

Utan den kan även tränas med data. Här kan vi se resultatet av att vi använder tre olika värden och gav den R^2 ≈ 0.089. vilket då innebär att den förklarar ca 8.9% av variationen i responsvariablen.
När vi kollar F- testet ser vi att p-värdenra är toklåga och kan därgenom uppskatta att alla bär en hög statistisk signifikans. Något som kan tala emot våran upptäckt är att förklarings graden är låg. Detta kan bero på ett antal saker, så som bristande data eg. variabler som saknas eller att det inte är statistiskt linjärt.  Modellen kan alltså inte anses vara bra för prediktion men visar ändå att vissa variabler påverkar utfallet.